system-addon/content-src/components/TopSites/TopSiteForm.jsx

```diff
@@ -102,33 +98,28 @@ export class TopSiteForm extends React.PureComponent {
 
     return (
       <form className="topsite-form">
-        <section className="edit-topsites-inner-wrapper">
+        <div className="form-input-container">
           <div className="form-wrapper">
             <h3 className="section-title">
               <FormattedMessage id={showAsAdd ? "topsites_form_add_header" : "topsites_form_edit_header"} />
             </h3>
-            <div className="field title">
-              <input
-                type="text"
-                value={this.state.label}
-                onChange={this.onLabelChange}
-                placeholder={this.props.intl.formatMessage({id: "topsites_form_title_placeholder"})} />
-            </div>
-            <div className={`field url${this.state.validationError ? " invalid" : ""}`}>
-              <input
-                type="text"
-                ref={this.onUrlInputMount}
-                value={this.state.url}
-                onChange={this.onUrlChange}
-                placeholder={this.props.intl.formatMessage({id: "topsites_form_url_placeholder"})} />
-              {this.state.validationError &&
-                <aside className="error-tooltip">
-                  <FormattedMessage id="topsites_form_url_validation" />
-                </aside>
-              }
-            </div>
+            <TopSiteFormInput onChange={this.onLabelChange}
+              value={this.state.label}
+              titleId="topsites_form_title_label"
+              placeholderId="topsites_form_title_placeholder"
+              errorMessageId="topsites_form_url_validation"
+              {...this.props} />
+            <TopSiteFormInput onChange={this.onUrlChange}
+              value={this.state.url}
+              onClear={this.onClearUrlClick}
+              validationError={this.state.validationError}
+              titleId="topsites_form_url_label"
+              placeholderId="topsites_form_url_placeholder"
+              errorMessageId="topsites_form_url_validation"
+              {...this.props} />
```

Comment:
What props are we expecting to pass on to the input?

Reply:
The only other prop needed was `intl`